File: app/src/main/java/com/rowland/delivery/features/dash/presentation/fragments/OverviewFragment.java
package com.rowland.delivery.features.dash.presentation.fragments;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;

import com.rowland.delivery.features.dash.presentation.activities.DashActivity;
import com.rowland.delivery.merchant.R;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;


public class OverviewFragment extends Fragment {

    private static final String TAB_POSITION = "tabposition";
    private int movePosition;
    private Unbinder unbinder;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @BindView(R.id.tabs)
    TabLayout tabLayout;

    @BindView(R.id.viewpager)
    ViewPager viewPager;


    public static OverviewFragment newInstance(int tabPosition) {
        OverviewFragment meals = new OverviewFragment();
        Bundle bundle = new Bundle();
        bundle.putInt(TAB_POSITION, tabPosition);

        meals.setArguments(bundle);
        return meals;
    }

    public OverviewFragment() {

    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            movePosition = getArguments().getInt(TAB_POSITION, 0);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_overview, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        unbinder = ButterKnife.bind(this, view);

        AppCompatActivity activity = ((AppCompatActivity) getActivity());
        DrawerLayout drawerLayout = ((DashActivity) activity).getDrawerLayout();

        if (toolbar != null) {
            activity.setSupportActionBar(toolbar);
            ActionBarDrawerToggle drawerToggle = new ActionBarDrawerToggle(activity, drawerLayout, toolbar, R.string.drawer_open, R.string.drawer_close) {
                @Override
                public void onDrawerSlide(View drawerView, float slideOffset) {
                    super.onDrawerSlide(drawerView, slideOffset);
                    this.syncState();
                    try {
                        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
                            Window window = getActivity().getWindow();
                            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
                            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
                            window.setStatusBarColor(getResources().getColor(R.color.colorPrimaryDarkTransparent));
                        }
                    } catch (Exception e) {
                        Log.e(OverviewFragment.class.getSimpleName(), ":" + e.fillInStackTrace().toString());
                    }
                }

                @Override
                public void onDrawerStateChanged(int newState) {
                    super.onDrawerStateChanged(newState);
                    this.syncState();
                }

                public void onDrawerClosed(View view) {
                    super.onDrawerClosed(view);
                    this.syncState();

                    try {
                        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
                            Window window = getActivity().getWindow();
                            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
                            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
                            window.setStatusBarColor(getResources().getColor(R.color.colorPrimaryDark));
                        }
                    } catch (Exception e) {
                        Log.e(OverviewFragment.class.getSimpleName(), ":" + e.fillInStackTrace().toString());
                    }
                }

                public void onDrawerOpened(View drawerView) {
                    super.onDrawerOpened(drawerView);
                    this.syncState();
                }
            };
            drawerLayout.addDrawerListener(drawerToggle);
            drawerToggle.syncState();
        }

        ViewPagerAdapter adapter = new ViewPagerAdapter(getChildFragmentManager());
        adapter.addFragment(new CategoryFragment(), getResources().getString(R.string.tab_categories));
        adapter.addFragment(new OrderFragment(), getResources().getString(R.string.tab_orders));
        viewPager.setAdapter(adapter);

        viewPager.setCurrentItem(movePosition);
        tabLayout.setupWithViewPager(viewPager);
    }

    @Override
    public void onDetach() {
        super.onDetach();
        unbinder.unbind();
    }


    private class ViewPagerAdapter extends FragmentStatePagerAdapter {

        private List<Fragment> mFragmentList;
        private final List<String> mFragmentTitleList;


        public ViewPagerAdapter(FragmentManager fm) {
            super(fm);
            mFragmentList = new ArrayList<>();
            mFragmentTitleList = new ArrayList<>();
        }

        public void addFragment(Fragment fragment, String title) {
            this.mFragmentList.add(fragment);
            this.mFragmentTitleList.add(title);
        }

        @Override
        public Fragment getItem(int position) {
            return mFragmentList.get(position);
        }

        @Override
        public int getCount() {
            return mFragmentList.size();
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return mFragmentTitleList.get(position);
        }
    }
}

File: app/src/main/java/com/rowland/delivery/features/dash/presentation/fragments/OverviewFragment.kt
package com.rowland.delivery.features.dash.presentation.fragments

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import com.rowland.delivery.features.dash.presentation.activities.DashActivity
import com.rowland.delivery.merchant.R
import kotlinx.android.synthetic.main.activity_dash.*
import kotlinx.android.synthetic.main.fragment_overview.*
import java.util.*


class OverviewFragment : Fragment() {
    private var movePosition: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            movePosition = arguments!!.getInt(TAB_POSITION, 0)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_overview, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val activity = activity as AppCompatActivity?
        val drawerLayout = (activity as DashActivity).dash_drawer_layout

        if (overview_toolbar != null) {
            activity.setSupportActionBar(overview_toolbar)
            val drawerToggle = object : ActionBarDrawerToggle(activity, drawerLayout, overview_toolbar, R.string.drawer_open, R.string.drawer_close) {
                override fun onDrawerSlide(drawerView: View, slideOffset: Float) {
                    super.onDrawerSlide(drawerView, slideOffset)
                    this.syncState()
                    try {
                        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
                            val window = getActivity()!!.window
                            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
                            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
                            window.statusBarColor = resources.getColor(R.color.colorPrimaryDarkTransparent)
                        }
                    } catch (e: Exception) {
                        Log.e(OverviewFragment::class.java.simpleName, ":" + e.fillInStackTrace().toString())
                    }

                }

                override fun onDrawerStateChanged(newState: Int) {
                    super.onDrawerStateChanged(newState)
                    this.syncState()
                }

                override fun onDrawerClosed(view: View) {
                    super.onDrawerClosed(view)
                    this.syncState()

                    try {
                        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
                            val window = getActivity()!!.window
                            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
                            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
                            window.statusBarColor = resources.getColor(R.color.colorPrimaryDark)
                        }
                    } catch (e: Exception) {
                        Log.e(OverviewFragment::class.java.simpleName, ":" + e.fillInStackTrace().toString())
                    }

                }

                override fun onDrawerOpened(drawerView: View) {
                    super.onDrawerOpened(drawerView)
                    this.syncState()
                }
            }
            drawerLayout.addDrawerListener(drawerToggle)
            drawerToggle.syncState()
        }

        val adapter = ViewPagerAdapter(childFragmentManager)
        adapter.addFragment(CategoryFragment.Companion.newInstance(null), resources.getString(R.string.tab_categories))
        adapter.addFragment(OrderFragment.Companion.newInstance(null), resources.getString(R.string.tab_orders))
        overview_viewpager.adapter = adapter

        overview_viewpager.currentItem = movePosition
        overview_tabs.setupWithViewPager(overview_viewpager)
    }

    private inner class ViewPagerAdapter(fm: androidx.fragment.app.FragmentManager) : FragmentStatePagerAdapter(fm) {

        private val mFragmentList: MutableList<androidx.fragment.app.Fragment>
        private val mFragmentTitleList: MutableList<String>


        init {
            mFragmentList = ArrayList()
            mFragmentTitleList = ArrayList()
        }

        fun addFragment(fragment: androidx.fragment.app.Fragment, title: String) {
            this.mFragmentList.add(fragment)
            this.mFragmentTitleList.add(title)
        }

        override fun getItem(position: Int): androidx.fragment.app.Fragment {
            return mFragmentList[position]
        }

        override fun getCount(): Int {
            return mFragmentList.size
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return mFragmentTitleList[position]
        }
    }

    companion object {

        private val TAB_POSITION = "tabposition"

        fun newInstance(tabPosition: Int): OverviewFragment {
            val frag = OverviewFragment()
            val bundle = Bundle()
            bundle.putInt(TAB_POSITION, tabPosition)

            frag.arguments = bundle
            return frag
        }
    }
}
